src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -336,6 +341,9 @@ private void checkEnvironmentProperties() {
     private void updateFromSpringBootPropertiesIfAny() {
         if (springbootEnvMap.containsKey(ARG_BLACKDUCK_URL)) {
             blackduckHost = springbootEnvMap.get(ARG_BLACKDUCK_URL);
+            if (blackduckHost != null && blackduckHost.contains(SYS_ENV_PROP_BLACKDUCK_URL)) {
+                blackduckHost = utility.getSysEnvProperty(SYS_ENV_PROP_BLACKDUCK_URL);
+            }
```

Comment:
This check has been added to address the scenario where customers could cross reference a system property as value in the Spring boot JSON property. For example: `export SPRING_APPLICATION_JSON='{"blackduck.url":"${BLACKDUCK_URL}"}'`

This scenario is currently not documented. @sig-cpotts